server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -187,6 +189,7 @@ private void finishRequest() {
                     () -> new WritePrimaryResult<>(
                         context.getBulkShardRequest(), context.buildShardResponse(), context.getLocationToSync(), null,
                         context.getPrimary(), logger));
+                primary.getBulkOperationListener().afterBulk(request.totalSizeInBytes(), System.nanoTime() - startBulkTime);
```

Comment:
This stats is important to reflect the processing power of bulk operations, and I feel that rejected bulk operations should not contribute to the time and size, because the processing power of bulk operations can become untrusted when there is a large number of rejections. So is it appropriate for me to put statistics at the end of the doRun interface ?